test/test_real_pebble.py

```diff
@@ -291,16 +300,22 @@ def test_log_forwarding(self):
         assert plan.log_targets["pretend-loki"].labels == {"foo": "bar"}
 
 
-@unittest.skipUnless(os.getenv('RUN_REAL_PEBBLE_TESTS'), 'RUN_REAL_PEBBLE_TESTS not set')
+@pytest.mark.skipif(
+    os.getenv('RUN_REAL_PEBBLE_TESTS') != '1',
+    reason='RUN_REAL_PEBBLE_TESTS not set',
+)
 class TestPebbleStorageAPIsUsingRealPebble(unittest.TestCase, PebbleStorageAPIsTestMixin):
-    def setUp(self):
+    @pytest.fixture
+    def prefix(self):
         pebble_path = os.getenv('PEBBLE')
         assert pebble_path is not None
-        self.prefix = tempfile.mkdtemp(dir=pebble_path)
-        self.client = pebble.Client(socket_path=get_socket_path())
+        prefix = tempfile.mkdtemp(dir=pebble_path)
+        yield prefix
+        shutil.rmtree(prefix)
```

Comment:
**env var**

I find it a little weird that the containing directory is taken from the environment. If it's to align to some code that uses real pebble, I still find it a little odd, because prefix is going to be a subdirectory of real pebble's dir.

**temp dir creation style**

pytest has a `tmpdir_factory` fixture that may be used instead, if the API allows sufficient customisation. I think there's a `root=` kwarg?

if not, perhaps the code could look like `with tempfile.NamedTemporaryDirectory(some_args) as prefix: ...` ?

Reply:
pytest has the tmpdir fixture, which works, but previously we had this discussion and we do not want to use a lot of fixtures in the arg list, that's why I kept the original tempfile and its logic here.